Full code pull request with corrected.

TheCoolerSuptelov opened the pull request.

TheCoolerSuptelov committed:
Add default server port
	modified:   src/main/resources/application.properties

TheCoolerSuptelov committed:
Table creation.
Class creation.
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	new file:   src/main/resources/db/migration/V1__initSchema.sql
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/resources/application.properties
	modified:   src/main/resources/db/migration/V1__initSchema.sql

TheCoolerSuptelov committed:
Add service class
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/main/resources/application.properties
	modified:   src/main/resources/db/migration/V1__initSchema.sql

TheCoolerSuptelov committed:
Add repo classes
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/main/resources/application.properties
	modified:   src/main/resources/db/migration/V1__initSchema.sql

TheCoolerSuptelov committed:
Add @CreationTimestamp
Add User creatin function in controller
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Fixed Sonar report issues
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Modif checkStyle issues
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MsgbackendApplicationTests.java

TheCoolerSuptelov committed:
Add user test. Change schema and hibernate entity naming strategy
	modified:   pom.xml
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/resources/application.properties
	modified:   src/main/resources/db/migration/V1__initSchema.sql
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MsgbackendApplicationTests.java
	new file:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java

TheCoolerSuptelov committed:
Made the field errorDetails optional on json output
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java

TheCoolerSuptelov committed:
Test improvements
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java

TheCoolerSuptelov committed:
Wrote integration test on user api
	modified:   pom.xml
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java
	new file:   src/test/resources/application.properties
	new file:   src/test/resources/docker-compose.yml

TheCoolerSuptelov committed:
Done rest api chat creation

	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java

TheCoolerSuptelov committed:
Fix table "Chat" structure. End with creation chat using list userNames
	modified:   README.md
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	new file:   src/main/resources/db/migration/V2__AddCreationAtToChat.sql

TheCoolerSuptelov committed:
Edite response chat and user controller
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java

TheCoolerSuptelov committed:
Failed to find chat and user in 1 query
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Add message controller logic.
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Rest Api method that return all chats concrete user.
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/resources/application.properties

TheCoolerSuptelov committed:
Add rest api method to get all messages from 1 paticular chat.
	modified:   README.md
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
Reorganize files
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java

TheCoolerSuptelov committed:
Refactor UserDto
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MsgbackendApplicationTests.java

TheCoolerSuptelov committed:
Refactor MessageDto
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
Refactor ChatDto
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java

TheCoolerSuptelov committed:
Add pagination
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java

TheCoolerSuptelov committed:
Add docker-compose for test environment
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java
	modified:   src/test/resources/application.properties
	modified:   src/test/resources/docker-compose.yml

TheCoolerSuptelov committed:
Added dockerFiles
	new file:   Dockerfile
	new file:   docker-compose.yml

TheCoolerSuptelov committed:
Reorganize chatService
	modified:   Dockerfile
	modified:   docker-compose.yml
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java
	modified:   src/test/resources/docker-compose.yml

TheCoolerSuptelov committed:
modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java

TheCoolerSuptelov committed:
Refactored unnecessary dependencies between services
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
Add exception. Not analyze input on services level, raise error instead.

	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java

TheCoolerSuptelov committed:
Rename variables. Delete "fromRequest"
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageController.java

TheCoolerSuptelov committed:
Change error handling in chatService. Instead of string with error description, throw errors
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java

TheCoolerSuptelov committed:
Change logic user search. Instead of hardCoded attribute names, use ENUM
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java

TheCoolerSuptelov committed:
modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java

TheCoolerSuptelov committed:
UserService now throw error then user trying create username with existed name.
Renamed test in TestUserApi.
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java

TheCoolerSuptelov committed:
Fix the ChatControllerTest.
Add flyway sql script test migration.
	new file:   src/main/resources/testdata/afterMigrate.sql
	new file:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	renamed:    src/test/java/com/github/thecoolersuptelov/msgbackend/TestUserApi.java -> src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java
	modified:   src/test/resources/application.properties

TheCoolerSuptelov committed:
Formatting
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java
	modified:   src/test/resources/application.properties

TheCoolerSuptelov committed:
Add message test cases
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/resources/testdata/afterMigrate.sql
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	new file:   src/test/java/com/github/thecoolersuptelov/msgbackend/MessageControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java
	modified:   src/test/resources/application.properties

TheCoolerSuptelov committed:
Fix sonar issues. Reformat code
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/MsgbackendApplication.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageCreationException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageDto.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MessageControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java

TheCoolerSuptelov committed:
Delete setters ID for Entities
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/User.java
	new file:   src/test/java/com/github/thecoolersuptelov/msgbackend/EndToEndApiTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MessageControllerTest.java

TheCoolerSuptelov committed:
Add end to end test
    modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/EndToEndApiTest.java

TheCoolerSuptelov committed:
Add test to chat creation with invalid user UUID
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatService.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/MessageControllerTest.java

TheCoolerSuptelov committed:
Reformat code
	modified:   src/test/java/com/github/thecoolersuptelov/msgbackend/ChatControllerTest.java

TheCoolerSuptelov committed:
modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
	new file:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageView.java

TheCoolerSuptelov committed:
Refactor response entity. Added exception handler
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserException.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java

TheCoolerSuptelov committed:
Added response status to exception handler

	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserController.java

andriilatyshev commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Column(name = "created_at")
+    @CreationTimestamp
+    private Timestamp created_at;
```
У тебя же есть аннотация, которая задает имя, зачем ты уничтожаешь джава код андерскор нотацией?

andriilatyshev commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/Chat.java:
```diff
@@ -0,0 +1,51 @@
+        return id;
+    }
+
+}
```
Уже не раз говорили о этой штуке.

andriilatyshev commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatController.java:
```diff
@@ -0,0 +1,42 @@
+        this.chatService = chatService;
+    }
+
+    @PostMapping(value = "addByUsername")
```
это по твоему по ресту?

andriilatyshev replied:
рест подразумевает, чт овсе есть ресурс, а действие над ресурсом определяет метод!

andriilatyshev commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chat/ChatRepository.java:
```diff
@@ -0,0 +1,22 @@
+
+    boolean existsByUsers_IdEqualsAndIdEquals(UUID userId, UUID chatId);
+
+    @Query(nativeQuery = true, value = "Select chat_users.chat_id as id, chat.name as name," + " chat.created_at as created_at, " + "max(COALESCE(message.created_at,to_timestamp(0))) from chat_users " + " inner join chat on chat.id = chat_users.chat_id " + "left join message on message.chat_id = chat_users.chat_id " + "where users_id = :userId " + "group by chat_users.chat_id,chat.name,chat.creat ...
```
не понимаю этого форматирования. Зачем конкатенация? В целом, я бы запросы выводил в константы, так читабельнее. Ну и в целом SQL тоже имеет форматирование и его лучше придерживаться. Оператор многострочной строки в этом поможет.

andriilatyshev commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java:
```diff
@@ -0,0 +1,88 @@
+        this.text = text;
+    }
+
+    public User getAuthor() {
```
почему ломбок не использовать?

andriilatyshev commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserSearchStrategy.java:
```diff
@@ -0,0 +1,13 @@
+    BY_ID("Id"), BY_USERNAME("UserName");
+
+    private final String searchAttribute;
+    private Object searchAlgo;
```
Что это и где используется?

andriilatyshev commented on src/main/java/com/github/thecoolersuptelov/msgbackend/chatUser/UserService.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+
+    public User addNewUser(UserDto userToCreation) throws UserException {
```
мы не используем в современной разработке чекед эксепшены.

andriilatyshev commented on src/test/java/com/github/thecoolersuptelov/msgbackend/UserControllerTest.java:
```diff
@@ -0,0 +1,47 @@
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class UserControllerTest {
```
не вижу негативных сценариев.

TheCoolerSuptelov committed:
Add one query creating message
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/Message.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageRepository.java
	modified:   src/main/java/com/github/thecoolersuptelov/msgbackend/chatMessage/MessageService.java
